src/hooks/useMultiSelect.stories.mdx

```diff
@@ -41,22 +39,134 @@ export default function MultiSelectWithHookExample() {
   const multiSelectId = "multi-select-1";
   // Without this hook, you'd have to add handlers for onChange and onClear,
   // and also manage the selectedItems state yourself.
-  const { onChange, onClear, selectedItems } = useMultiSelect(
-    multiSelectId,
-    items
+  const { onChange, onClear, selectedItems } = useMultiSelect();
+
+  return (
+    <MultiSelect
+      id={multiSelectId}
+      variant="listbox"
+      items={items}
+      selectedItems={selectedItems}
+      onChange={(selectedItem) => onChange(selectedItem.id, multiSelectId)}
+      onClear={() => {
+        onClear(multiSelectId);
+      }}
+    />
   );
+}
+```
+
+## useMultiSelect Hook implementation in a `dialog` MultiSelect
+
+```jsx
+import React from "react";
+import MultiSelect from "./MultiSelect";
+import useMultiSelect from "./MultiSelect/useMultiSelect";
+
+export default function MultiSelectWithHookExample() {
+  const multiSelectId = "multi-select-1";
+  // Without this hook, you'd have to add handlers for onChange and onClear,
+  // and also manage the selectedItems state yourself.
+  const { onChange, onMixedStateChange, onClear, selectedItems } =
+    useMultiSelect();
 
   return (
     <MultiSelect
       id={multiSelectId}
       variant="listbox"
       items={items}
       selectedItems={selectedItems}
-      onChange={(selectedItem) => onChange(selectedItem.id)}
+      onChange={(selectedItem) => onChange(selectedItem.id, multiSelectId)}
+      onMixedStateChange={(selectedItem) =>
+        onMixedStateChange(selectedItem.id, multiSelectId, items)
+      }
```

Comment:
The `variant` should be "dialog" and not "listbox". For `onChange` and `onMixedStateChange`, the callback functions should be `(e) => ....` since this takes the `e.target.id` for the "dialog" variant.

Reply:
fixed